src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,44 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  static checkHealth() {
+    // Update animalAlive to include only animals with health > 0
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(prey) {
+    if (prey instanceof Herbivore && !prey.hidden) {
+      prey.health -= 50; // Reduz saúde do herbívoro
+      Animal.checkHealth(); // Atualiza a lista de animais vivos
+    }
```

Comment:
In the `bite` method, the health of the `prey` is reduced by 50. However, there is no check to ensure that the `prey`'s health does not drop below zero. Consider adding a condition to set `prey.health` to zero if it becomes negative, to prevent negative health values.